### ISSUE TITLE
managed node pool without autoscaling broken in 1.129.0

### ISSUE THREAD

**Issue by aschleifer** [aliyun/terraform-provider-alicloud#3821]
#3772  changes the logic for detecting `scaling_config` from `d.HasChange("scaling_config")` to `d.GetOk("scaling_config")` (https://github.com/aliyun/terraform-provider-alicloud/pull/3772/commits/1c546bb780f408e4bc15a0e30b915f6f8f7d933e#diff-458e10746ce5c75ef4b86ede6c9a797e652b473dcdbc30fca6917189919d516fL633-R632). This introduces the bug that any change to a managed node pool where autoscaling was disabled before will now always enable autoscaling. This will happen with "0 values":

```
    scaling_config {
        eip_bandwidth = 0
        is_bond_eip   = false
        max_size      = 0
        min_size      = 0
    }
```

As these are the values which are written to the terraform state for managed node pools with autoscaling disabled. This then restults in the cluster breaking as the pool scales down to 0 nodes.

**Comment by tiankai66** [comment]
Your feedback has been received. We will process it as soon as possible.
Thank you for your trust and support of Alibaba Cloud!

**Comment by anserhyb** [comment]
If autoscaling is turned off, then you need to remove the configuration scaling_config from the .tf configuration file

**Comment by anserhyb** [comment]
This change was rolled back, but I fixed it in version 1.152.0. To avoid re-enabling autoscaling, you need to comment out or delete the `scaling_config` configuration in the `.tf` file before running apply. for example:
```
resource "alicloud_cs_kubernetes_node_pool" "default" {
  
  # other parameters ...

  ## autoscaling will not be enabled after commenting out scaling_config
  # scaling_config {
  #   min_size      = 2
  #   max_size      = 20  
  # }
  
}
```

**Comment by aschleifer** [comment]
this bug was solved, but just as a side note: once disabled autoscaling can't be re-enabled on a node-pool without rebuilding it.
